### PR TITLE
Master kpfm

### LINKED ISSUE #43: Merging the KPFM part to the main.
Aurelio will need to compare the latest development of KPFM in here https://github.com/aureliojgc/ProbeParticleModel/tree/OpenCL_KPFM_py3

### PR DESCRIPTION
**Pull request by aureliojgc**
Fixes #43 

Request for pull the kpfm modifications.
Basically I add the cycles over bias in generateELFF, both the relaxed_scan and relaxed_scan_PVE and the plot_results.
I made relaxed_scan_PVE to work also with LJ
The rest are minor changes around the kpfm needs.

I have tested all AFM and KPFM with density tips and orbital-like tips for AIMS and VASP.

### CODE REVIEW

File: pyProbeParticle/HighLevel.py [hunk 1]
```diff
@@ -212,7 +216,7 @@ def computeELFF_pointCharge( geomFile, tip='s', save_format=None, computeVpot=Fa
     if(verbose>0): print("<<<END: computeELFF_pointCharge()")
     return FF, V, nDim, lvec
 
-def computeElFF(V,lvec,nDim,tip,computeVpot=False, tilt=0.0 ):
+def computeElFF(V,lvec,nDim,tip,computeVpot=False, tilt=0.0,sigma=PPU.params['sigma'] ):
```
Comment (NikoOinonen) [inline pyProbeParticle/HighLevel.py:219]: This is probably not doing what you think it does. The function signature gets evaluated only once when the module is imported, so the `sigma` value here will not get updated even if the value in `PPU.params['sigma']` changes. It's probably better if you set the default value to `None` and then inside function check if it's `None` and set it to `PPU.params['sigma']` if it is.

Comment (aureliojgc) [reply pyProbeParticle/HighLevel.py]: Hi Nico,
Thanks for the review. I rewrote the function acording to your comments and tested it againg. I think it is correct now.

Comment (NikoOinonen) [reply pyProbeParticle/HighLevel.py]: Ok, looks good.

**Review by ProkopHapala — APPROVED** [review]
seems fine, assuming you tested original functionality works (I didn't)

**Review by ProkopHapala — CHANGES_REQUESTED** [review]
Seems fine, assuming you tested it. 

Consider if you really need to copy V and rho before you subtract it? Do you need to save original array? Are you worried that the original array may be changed? It can cost quite a lot of memory if the arrays are large.

**Review by ProkopHapala — APPROVED** [review]

**Review by ProkopHapala — APPROVED** [review]
seems fine assuming you tested that original functionality works (I didn't)

File: relaxed_scan.py [hunk 2]
```diff
@@ -69,12 +73,26 @@ def rotFF( Fx,Fy, a ):
...
         if ( abs(Q) > 1e-7):
             charged_system=True
+    # Vkpfm
+    aplied_bias=False
+    if opt_dict['Vrange'] is not None:
+        Vs = np.linspace( opt_dict['Vrange'][0], opt_dict['Vrange'][1], int(opt_dict['Vrange'][2]) )
+    elif opt_dict['Vbias'] is not None:
+        Vs = [ opt_dict['Vbias'] ]
+    else:
+        Vs = [0.0]
```
Comment (ProkopHapala) [inline relaxed_scan.py:83]: OK, so it now always loop over Vs, but when KPFM is inactive Vs=[0.0]

**Review by ProkopHapala — APPROVED** [review]
seems OK, as long as it is tested

**Pull request merged**